gaia_tasks/inputs.py

```diff
@@ -0,0 +1,187 @@
+        #             headers = {'Authorization': basic_auth}
+
+        #         with open(self.uri, 'w') as outjson:
+        #             r = requests.get(url, headers=headers)
+        #             r.raise_for_status()
+        #             json.dump(r.json(), outjson)
+        #     else:
+        #         raise GaiaException('This server {} is not supported. \n{}'.format(minerva))
+        else:
+            raise GaiaException('Unsupported data source. \n{}'.format(minerva))
+
+    def read(self, epsg=None, **kwargs):
+        """
+        Read vector data from Girder
+        :param format: Format to return data in (default is GeoDataFrame)
+        :param epsg: EPSG code to reproject data to
+        :return: Data in GeoJSON
+        """
+
+        if self.data is None:
+            self.save_geojson()
+            self.data = geopandas.read_file(self.uri)
+            if self.filters:
+                self.filter_data()
+        out_data = self.data
+        if epsg and self.get_epsg() != epsg:
+            out_data = geopandas.GeoDataFrame.copy(out_data)
+            out_data[out_data.geometry.name] = \
+                self.data.geometry.to_crs(epsg=epsg)
+            out_data.crs = fiona.crs.from_epsg(epsg)
+        if format == formats.JSON:
+            return out_data.to_json()
+        else:
+            return out_data
+
+    def write(self, filename=None, as_type='json'):
+        """
+        Write data (assumed geopandas) to geojson or shapefile
+        :param filename: Base filename
+        :param as_type: json or memory
+        :return: file girder uri
+        """
+        filedata = self.data.to_json()
+        if not filename:
+            filename = self.filename
+        if as_type == 'json':
+            self.uri = self.uri.replace(os.path.basename(self.uri), filename)
+            self.create_output_dir(self.uri)
+            with open(self.uri, 'w') as outfile:
+                outfile.write(filedata)
+        elif as_type == 'memory':
+            pass
+        else:
+            raise NotImplementedError('{} not a valid type'.format(as_type))
+
+        fd = StringIO(filedata)
+        upload = self.client.uploadFile(parentId=self.id, stream=fd,
+                                        size=len(filedata), name=filename)
+        item_meta = self.client.getItem(self.id)['meta']
+        item_meta['minerva']['geojson_file'] = {
+            '_id': upload['_id'],
+            'name': upload['name']
+        }
+        item_meta['minerva']['geo_render'] = {
+            'type': 'geojson',
+            'file_id': upload['_id']
+        }
+        self.client.addMetadataToItem(self.id, item_meta)
+        return os.path.join(
+            self.client.urlBase, 'file', upload['_id'], 'download')
+
+    def filter_data(self):
+        """
+        Apply filters to the dataset
+        :return:
+        """
+        self.data = filter_pandas(self.data, self.filters)
```

Comment:
In the spirit of new style girder_worker tasks, this class should really be implemented as a transform like those defined in [girder_io.py](https://github.com/girder/girder_worker_utils/blob/master/girder_worker_utils/transforms/girder_io.py).  Much of what you are doing here is already implemented in the existing transforms.  I don't have a clear sense at the moment whether or not this is an easy thing to do (or even makes any sense) given the current API in Gaia.